x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/new_index/method_api.tsx

```diff
@@ -8,46 +8,37 @@
 /**
  * TODO:
  * - Need to add documentation URLs (search for `#`s)
- * - Need get dynamic Enterprise Search API URL
- * - Port over existing API view from App Search to the panel below.
- * - move the endpoint state to a logic file
- * - Replace `onNameChange` logic with that from App Search
  * - Need to implement the logic for the attaching search engines functionality
  */
 
-import React, { useState } from 'react';
+import React from 'react';
 
-import { EuiCode, EuiLink, EuiPanel, EuiSpacer, EuiText, EuiTitle } from '@elastic/eui';
+import { useValues } from 'kea';
+
+import { EuiCodeBlock, EuiLink, EuiPanel, EuiSpacer, EuiText, EuiTitle } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
 import { FormattedMessage } from '@kbn/i18n-react';
 
+import { getEnterpriseSearchUrl } from '../../../shared/enterprise_search_url/external_url';
+
+import { DOCUMENTS_API_JSON_EXAMPLE } from './constants';
+import { NewSearchIndexLogic } from './new_search_index_logic';
 import { NewSearchIndexTemplate } from './new_search_index_template';
 
 export const MethodApi: React.FC = () => {
-  const [endpoint, setEndpoint] = useState('');
+  const { name } = useValues(NewSearchIndexLogic);
+  const apiKey = 1212312313; // TODO change this
```

Comment:
Is this a follow-up PR todo?

Reply:
yes, as there are some changes. Will keep it as it for now